src/MOI_wrapper.jl

```diff
@@ -176,6 +176,59 @@ function MOI.optimize!(model::Optimizer)
 end
 
 
+const QP_SET_TYPES = (
+    MOI.GreaterThan{Float64},
+    MOI.LessThan{Float64},
+    MOI.EqualTo{Float64},
+    MOI.Interval{Float64}
+)
+
+const QP_FUNCTION_TYPES = (
+    MOI.SingleVariable,
+    MOI.ScalarAffineFunction{Float64}
+)
+
+const QP_OBJECTIVE_TYPES = (
+    MathOptInterface.ScalarAffineFunction{Float64},
+    MathOptInterface.ScalarQuadraticFunction{Float64},
+)
+
+"""
+    backward(model::Optimizer, params...)
+
+Wrapper method for the backward pass. Checks the model/problem type and dispatches a call to
+[`backward_quad`](@ref) or [`backward_conic`](@ref) accordingly.
+"""
+function backward(model::Optimizer, params...)
+    # check if the model is a QP
+    isQP = true
+
+    # check constraints
+    con_types = MOI.get(model.optimizer, MOI.ListOfConstraints())
+    for types in con_types
+        func = types[1]
+        set = types[2]
+
+        if !in(func, QP_FUNCTION_TYPES) || !in(set, QP_SET_TYPES)
+            isQP = false
+        end
+    end
+
+    # check objective
+    # works both for affine and quadratic objective functions
+    obj = MOI.get(model.optimizer, MOI.ObjectiveFunction{MOI.ScalarQuadraticFunction{Float64}}())
```

Comment:
You should do `MOI.get(model.optimizer, MOI.ObjectiveFunctionType()) <: QP_OBJECTIVE_TYPES` instead.

Reply:
Here, you as for a quadratic objective so it will either error or give you a quadratic objective.